libp2p/stream/bufferstream.nim

```diff
@@ -71,12 +73,12 @@ method pushData*(s: BufferStream, data: seq[byte]) {.base, async.} =
 
   # We will block here if there is already data queued, until it has been
   # processed
-  inc s.pushing
   try:
+    s.pushing = true
```

Comment:
Can't say I'm happy to see that much bool flags and logic really...
We are a single threaded polled in order app ideally.
But if works let's do it, I don't see any flaw. Have to test tho properly.